### PR TITLE
Enable deadcode, goconst, goimports and golint linters and apply their fixes

### PR DESCRIPTION
**Pull request by saschagrunert**
Needs https://github.com/containers/storage/pull/521

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by saschagrunert** [comment]
Rebased on top of the latest master.

**Comment by saschagrunert** [comment]
> LGTM
> 
> Currently, we're still not using a blacklist. I'd love to enable all linters and selectively disable the ones that break in the disable list.

Ah okay, like this: https://github.com/containers/storage/pull/522/commits/18e6f816dbb4e0799d7c4bcfcb05ebdf0774cc65 ?

**Comment by vrothberg** [comment]
> > LGTM
> > Currently, we're still not using a blacklist. I'd love to enable all linters and selectively disable the ones that break in the disable list.
> 
> Ah okay, like this: [18e6f81](https://github.com/containers/storage/commit/18e6f816dbb4e0799d7c4bcfcb05ebdf0774cc65) ?

Yes, exactly. Thanks! This way, when updating to a newer version, we automatically pull in new linters.

File: images.go [hunk 1]
```diff
@@ -214,17 +214,17 @@ func bigDataNameIsManifest(name string) bool {
 
 // recomputeDigests takes a fixed digest and a name-to-digest map and builds a
 // list of the unique values that would identify the image.
-func (image *Image) recomputeDigests() error {
-	validDigests := make([]digest.Digest, 0, len(image.BigDataDigests)+1)
+func (i *Image) recomputeDigests() error {
```
Comment (TomSweeneyRedHat) [inline images.go]: Ugh, I much prefer image.  Searching for 'i' hits on so many finds it's hard to parse, even on a small chunk of code like this.

File: drivers/driver.go [hunk 2]
```diff
@@ -48,9 +48,9 @@ type CreateOpts struct {
 type MountOpts struct {
 	// Mount label is the MAC Labels to assign to mount point (SELINUX)
 	MountLabel string
-	// UidMaps & GidMaps are the User Namespace mappings to be assigned to content in the mount point
-	UidMaps []idtools.IDMap
-	GidMaps []idtools.IDMap
+	// UIDMaps & GIDMaps are the User Namespace mappings to be assigned to content in the mount point
+	UIDMaps []idtools.IDMap
+	GIDMaps []idtools.IDMap
```
Comment (vrothberg) [inline drivers/driver.go]: API break

File: pkg/mount/mounter_linux.go [hunk 3]
```diff
@@ -13,14 +13,16 @@ const (
 
 	// broflags is the combination of bind and read only
 	broflags = unix.MS_BIND | unix.MS_RDONLY
+
+	none = "non"
```
Comment (vrothberg) [inline pkg/mount/mounter_linux.go]: Typo: the e is missing

File: drivers/driver.go (hunk 2 above)
Comment (vrothberg) [reply drivers/driver.go]: I suggest reverting these specific changes and instruct the linter to ingore it.

**Comment by rhatdan** [comment]
Oops, I agree, @saschagrunert Could you open a PR to fix these two issues.

File: pkg/mount/mounter_linux.go (hunk 3 above)
Comment (saschagrunert) [reply pkg/mount/mounter_linux.go]: Fixed in https://github.com/containers/storage/pull/523

File: drivers/driver.go (hunk 2 above)
Comment (saschagrunert) [reply drivers/driver.go]: Fixed in https://github.com/containers/storage/pull/523

**Comment by saschagrunert** [comment]
Sure, I opened: https://github.com/containers/storage/pull/523
